submissions/pozzitive11/js-world-oop/index.js

```diff
@@ -0,0 +1,75 @@
+class Inhabitant {
+  constructor(species, name, gender, saying) {
+    this.species = species;
+    this.name = name;
+    this.gender = gender;
+    this.saying = saying;
+  }
+
+  getProperties() {
+    return `${this.species}; My name ${this.name}; ${this.gender}; ${this.saying}`;
+  }
+}
+class Animal extends Inhabitant {
+  constructor(species, name, gender, paw, saying) {
+    super(species, name, gender, saying);
+    this.paw = paw;
+  }
+  getProperties() {
+    return super.getProperties() + `; I have ${this.paw} paws;`;
+  }
+}
+
+class Human extends Inhabitant {
+  constructor(species, name, gender, saying) {
+    super(species, name, gender, saying);
+    this.legs = 2;
+    this.hands = 2;
+  }
+
+  getProperties() {
+    return (
+      super.getProperties() +
+      `; I have ${this.legs} legs and ${this.hands} hands;`
+    );
+  }
+}
+
+class Dog extends Animal {
+  constructor(species, name, gender, saying, paw) {
+    super(species, name, gender, paw, saying);
```

Comment:
When we create an instance of `Dog` why would we need to specify the species?
`new Dog('dog')` seems redundant. Same refers to `Cat` and `Human`.

> You may pass scalar values to a super's constructor, not just simply hand over some params.